addons/payment_xendit/static/src/js/payment_form.js

```diff
@@ -0,0 +1,199 @@
+            return; // Don't re-extract the data if already done for this payment method.
+        }
+
+        // Overwrite the flow of the selected payment method.
+        this._setPaymentFlow('direct');
+
+        // Extract and deserialize the inline form values.
+        const radio = document.querySelector('input[name="o_payment_radio"]:checked');
+        const inlineForm = this._getInlineForm(radio);
+        const xenditForm = inlineForm.querySelector('[name="o_xendit_form"]');
+        this.xenditData[paymentOptionId] = JSON.parse(
+            xenditForm.dataset['xenditInlineFormValues']
+        );
+        let xenditURL = 'https://js.xendit.co/v1/xendit.min.js';
+        this.xenditData[paymentOptionId].form = xenditForm;
+        this.xenditData[paymentOptionId].xenditURL = xenditURL;
+
+        // Load the SDK.
+        loadJS(xenditURL);
+    },
+
+    // #=== PAYMENT FLOW ===#
+
+    /**
+     * Trigger the payment processing by submitting the data.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {string} providerCode - The code of the selected payment option's provider.
+     * @param {number} paymentOptionId - The id of the selected payment option.
+     * @param {string} paymentMethodCode - The code of the selected payment method, if any.
+     * @param {string} flow - The payment flow of the selected payment option.
+     * @return {void}
+     */
+    async _initiatePaymentFlow(providerCode, paymentOptionId, paymentMethodCode, flow) {
+        if (providerCode !== 'xendit' || flow === 'token' || paymentMethodCode != 'card') {
+            this._super(...arguments); // Tokens are handled by the generic flow
+            return;
+        }
+        const inputs = Object.values(
+            this._xenditGetInlineFormInputs(paymentOptionId, paymentMethodCode)
+        );
+        // Checking the elements
+        if (!inputs.every(element => element.reportValidity())) {
+            this._enableButton(); // The submit button is disabled at this point, enable it
+            return;
+        }
+
+        await this._super(...arguments);
+    },
+
+    /**
+     * Process the direct payment flow.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {string} providerCode - The code of the selected payment option's provider.
+     * @param {number} paymentOptionId - The id of the selected payment option.
+     * @param {string} paymentMethodCode - The code of the selected payment method, if any.
+     * @param {object} processingValues - The processing values of the transaction.
+     * @return {void}
+     */
+    async _processDirectFlow(providerCode, paymentOptionId, paymentMethodCode, processingValues) {
+        if (providerCode !== 'xendit') {
+            this._super(...arguments);
+            return;
+        }
+        // if tokenize, create multi-use token
+        const payload = {
```

Comment:
`currency` should be passed to Xendit to avoid discrepancies between the platforms. According to [Xendit's documentation](https://developers.xendit.co/api-reference/#create-token), if `currency` is not given, it defaults based on your business country.